torch/csrc/jit/mobile/model_compatibility.cpp

```diff
@@ -0,0 +1,107 @@
+#include <ATen/core/ivalue.h>
+#include <caffe2/serialize/file_adapter.h>
+#include <caffe2/serialize/inline_container.h>
+#include <torch/csrc/jit/api/compilation_unit.h> // removed after using simple type_resolver/obj_loader
+#include <torch/csrc/jit/mobile/import.h> // removed after using simple type_resolver/obj_loader
+#include <torch/csrc/jit/mobile/model_compatibility.h>
+#include <torch/csrc/jit/serialization/import_read.h>
+
+#include <string>
+#include <vector>
+
+namespace torch {
+namespace jit {
+
+using caffe2::serialize::FileAdapter;
+using caffe2::serialize::IStreamAdapter;
+using caffe2::serialize::PyTorchStreamReader;
+using caffe2::serialize::ReadAdapterInterface;
+
+namespace {
+
+c10::IValue readArchive(
+    const std::string& archive_name,
+    PyTorchStreamReader& stream_reader) {
+  c10::optional<at::Device> device;
+  std::shared_ptr<CompilationUnit> compilation_unit =
+      std::make_shared<CompilationUnit>();
+
+  // TODO (T90180710): Simplify type_resolver and obj_loader when getting
+  // bytecode version from model
+  auto type_resolver = [&](const c10::QualifiedName& qn) {
+    return typeResolverGeneral(qn, compilation_unit);
+  };
+
+  std::shared_ptr<mobile::CompilationUnit> mobile_compilation_unit =
+      std::make_shared<mobile::CompilationUnit>();
+  auto obj_loader = [&](at::StrongTypePtr type, IValue input) {
+    return objLoaderGeneral(type, input, mobile_compilation_unit);
+  };
+
+  auto ivalues = torch::jit::readArchiveAndTensors(
+      archive_name, type_resolver, obj_loader, device, stream_reader);
+  return ivalues;
+}
+
+bool check_zip_file(std::shared_ptr<ReadAdapterInterface> rai) {
```

Comment:
Is it necessary to do this check? I don't think we are doing this for mobile load. Also is it copied from serialization/import.cpp? If so, would be good to share the code and comments.

Reply:
Move the one in `serialization/import.cpp` to `serialization/import_read`, and reuse it here.